D1673/P1673.md

```diff
@@ -6601,103 +6673,110 @@ void test_conjugate_transposed(
 concept @_in-vector_@ = // exposition only
   @_is-mdspan_@<T>::value &&
   T::rank() == 1;
-```
-
-* `in_vector*_t` is a rank-1 `mdspan` with a
-    potentially `const` element type and a unique layout.
-    If the algorithm accesses the object, it will do so in read-only fashion.  (TODO MOVE THIS LAST SENTENCE TO A COMMON PLACE)
 
-```c++
 template<class T>
-concept @_inout-vector_@ = // exposition only
+concept @_out-vector_@ = // exposition only
   @_is-mdspan_@<T>::value &&
   T::rank() == 1 &&
   is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
   T::is_always_unique();
-```
 
-```c++
 template<class T>
 concept @_inout-vector_@ = // exposition only
   @_is-mdspan_@<T>::value &&
   T::rank() == 1 &&
   is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
   T::is_always_unique();
+
+template<class T>
+concept @_in-matrix_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 2;
+
+template<class T>
+concept @_out-matrix_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 2 &&
+  is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
+  T::is_always_unique();
+
+template<class T>
+concept @_inout-matrix_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 2 &&
+  is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
+  T::is_always_unique();
+
+template<class T>
+concept @_in-object_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  (T::rank() == 1 || T::rank() == 2);
+
+template<class T>
+concept @_out-object_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  (T::rank() == 1 || T::rank() == 2) &&
+  is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
+  T::is_always_unique();
+
+template<class T>
+concept @_inout-object_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  (T::rank() == 1 || T::rank() == 2) &&
+  is_same_v<remove_const_t<typename T::element_type>, typename T::element_type> &&
+  T::is_always_unique();
 ```
 
-* `out_vector*_t` is a rank-1 `mdspan`
-    with a non-`const` element type,
-    and whose layout is always unique
-    (`layout_type::is_always_unique()` equals `true`).
-    If the algorithm accesses the object, it will do so in write-only fashion.
+If an algorithm in *[linalg.algs]* accesses the elements
+of an _`in-vector`_, `in-matrix`, or `in-object`,
+it will do so in read-only fashion.
+
+*[Note:* The `element_type` of an
+_`in-vector`_, _`in-matrix`_, or _`in-object`_ need not be `const`,
+because an mdspan accessor with a nonconst `element_type`
+need not necessarily be convertible to an accessor
+with a const `element_type`. --*end note]*
 
-* `in_matrix*_t` is a rank-2 `mdspan` with a
-    `const` element type.
-    If the algorithm accesses the object, it will do so in read-only fashion.
+If an algorithm in *[linalg.algs]* accesses the elements
+of an _`out-vector`_, `out-matrix`, or `out-object`,
```

Comment:
Did you mean to only have `out-vector` in italic, ie not `out-matrix` and `out-object`?

Reply:
All these _`out-vector`_, _`out-matrix`_, and _`out-object`_ need to be italicized.  Thanks for pointing this out!

Reply:
Fixed; thanks!